servers/quarkus-common/src/main/java/org/projectnessie/quarkus/providers/storage/CacheInvalidationReceiver.java

```diff
@@ -0,0 +1,172 @@
+  private final Set<String> validTokens;
+  private final String invalidationPath;
+  private final ObjectMapper objectMapper;
+
+  @Inject
+  public CacheInvalidationReceiver(
+      QuarkusStoreConfig storeConfig,
+      DistributedCacheInvalidationHolder invalidationHolder,
+      @ServerInstanceId String serverInstanceId) {
+    this.invalidator = invalidationHolder;
+    this.serverInstanceId = serverInstanceId;
+    this.invalidationPath = storeConfig.cacheInvalidationUri();
+    this.validTokens =
+        new HashSet<>(storeConfig.cacheInvalidationValidTokens().orElse(emptyList()));
+    this.objectMapper =
+        new ObjectMapper()
+            // forward compatibility
+            .disable(FAIL_ON_UNKNOWN_PROPERTIES);
+  }
+
+  public void registerManagementRoutes(@Observes ManagementInterface mi) {
+    mi.router().post(invalidationPath).handler(this::cacheInvalidations);
+  }
+
+  void cacheInvalidations(RoutingContext rc) {
+    HttpServerRequest request = rc.request();
+    String senderId = request.getParam("sender");
+    String token = request.getHeader(NESSIE_CACHE_INVALIDATION_TOKEN_HEADER);
+
+    cacheInvalidations(
+        rc,
+        () -> {
+          try {
+            String json = rc.body().asString();
+            if (json == null || json.isEmpty()) {
+              return CacheInvalidations.cacheInvalidations(emptyList());
+            }
+            return objectMapper.readValue(json, CacheInvalidations.class);
+          } catch (Exception e) {
+            throw new RuntimeException(e);
+          }
+        },
+        senderId,
+        token);
+  }
+
+  void cacheInvalidations(
+      RoutingContext rc,
+      Supplier<CacheInvalidations> invalidations,
+      String senderId,
+      String token) {
+    if (token == null || !validTokens.contains(token)) {
+      LOGGER.warn("Received cache invalidation with invalid token {}", token);
+      responseInvalidToken(rc);
+      return;
+    }
+    if (serverInstanceId.equals(senderId)) {
+      LOGGER.trace("Ignoring invalidations from local instance");
+      responseNoContent(rc);
+      return;
+    }
+    if (!"application/json".equals(rc.request().getHeader("Content-Type"))) {
+      LOGGER.warn("Received cache invalidation with invalid HTTP content type");
+      responseInvalidContentType(rc);
+      return;
+    }
+
+    List<CacheInvalidations.CacheInvalidation> invs;
+    try {
+      invs = invalidations.get().invalidations();
+    } catch (RuntimeException e) {
+      responseServerError(rc);
+      return;
+    }
+
+    for (CacheInvalidations.CacheInvalidation invalidation : invs) {
+      switch (invalidation.type()) {
+        case CacheInvalidations.CacheInvalidationEvictObj.TYPE:
+          CacheInvalidations.CacheInvalidationEvictObj putObj =
+              (CacheInvalidations.CacheInvalidationEvictObj) invalidation;
+          invalidator.invalidation.evictObj(putObj.repoId(), objIdFromByteArray(putObj.id()));
```

Comment:
I'm a bit confused... If we have two servers, one with distributed invalidations enabled and the other disabled. When the second server receives an eviction event, won't `invalidator.invalidation` be `null`?